worlds/wargroove/Rules.py

```diff
@@ -0,0 +1,161 @@
+    set_rule(world.get_location('Best Friendssss: Victory', player), lambda state: True)
+    set_region_exit_rules(world.get_region('Best Friendssss', player),
+                          [world.get_location('Best Friendssss: Victory', player)])
+
+    set_rule(world.get_location('A Knight\'s Folly: Caesar', player), lambda state: True)
+    set_rule(world.get_location('A Knight\'s Folly: Victory', player), lambda state: True)
+    set_region_exit_rules(world.get_region('A Knight\'s Folly', player),
+                          [world.get_location('A Knight\'s Folly: Victory', player)])
+
+    set_rule(world.get_location('Denrunaway: Chest', player), lambda state: True)
+    set_rule(world.get_location('Denrunaway: Victory', player), lambda state: True)
+    set_region_exit_rules(world.get_region('Denrunaway', player), [world.get_location('Denrunaway: Victory', player)])
+
+    # Levels 3AA-3AC
+    set_rule(world.get_location('Dragon Freeway: Victory', player),
+             lambda state: state._wargroove_has_item(player, 'Mage'))
+    set_region_exit_rules(world.get_region('Dragon Freeway', player),
+                          [world.get_location('Dragon Freeway: Victory', player)])
+
+    set_rule(world.get_location('Deep Thicket: Find Sedge', player),
+             lambda state: state._wargroove_has_item(player, 'Mage'))
+    set_rule(world.get_location('Deep Thicket: Victory', player),
+             lambda state: state._wargroove_has_item(player, 'Mage'))
+    set_region_exit_rules(world.get_region('Deep Thicket', player),
+                          [world.get_location('Deep Thicket: Victory', player)])
+
+    set_rule(world.get_location('Corrupted Inlet: Victory', player),
+             lambda state: state._wargroove_has_item(player, 'Barge') or
+                           state._wargroove_has_item(player, 'Merfolk') or
+                           state._wargroove_has_item(player, 'Warship'))
+    set_region_exit_rules(world.get_region('Corrupted Inlet', player),
+                          [world.get_location('Corrupted Inlet: Victory', player)])
+
+    # Levels 3BA-3BC
+    set_rule(world.get_location('Mage Mayhem: Caesar', player),
+             lambda state: state._wargroove_has_item(player, 'Harpy') or state._wargroove_has_item(player, 'Dragon'))
+    set_rule(world.get_location('Mage Mayhem: Victory', player),
+             lambda state: state._wargroove_has_item(player, 'Harpy') or state._wargroove_has_item(player, 'Dragon'))
+    set_region_exit_rules(world.get_region('Mage Mayhem', player), [world.get_location('Mage Mayhem: Victory', player)])
+
+    set_rule(world.get_location('Endless Knight: Victory', player),
+             lambda state: state._wargroove_has_item(player, 'Eastern Bridges') and (
+                           state._wargroove_has_item(player, 'Spearman') or
+                           state._wargroove_has_item(player, 'Harpy') or
```

Comment:
I added a logic change here based on feedback. Spearmen, harpies or dragons can beat this level so long as the player has the Eastern Bridges. Previously, it was just spearmen.